### ISSUE TITLE
no-mutable-public-property-of-construct: properties without a type annotation are never reported

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#541]
### Describe the bug

`no-mutable-public-property-of-construct` never reports public properties that have **no explicit type annotation**, because the shared finder `findPublicPropertiesInClass` skips any `PropertyDefinition` / `TSParameterProperty` without `typeAnnotation`. The annotation requirement makes sense for `no-construct-in-public-property-of-construct` (which inspects the annotated type), but for the readonly check it is unrelated — mutability does not depend on whether the type is written out.

**Repro:**

```ts
import { Construct } from "constructs";

export class NoAnnotation extends Construct {
  public annotated: number = 0; // control: reported
  public inferred = 0;          // not reported
}
```

### Expected Behavior

Both properties are mutable public properties and should be reported.

### Current Behavior

```
repro/no-mutable-public-property-of-construct/no-annotation.ts  5:3  error  Public property 'annotated' should be readonly. Consider adding the 'readonly' modifier  awscdk/no-mutable-public-property-of-construct
```

`public inferred = 0;` is not reported. Oxlint behaves identically.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), macOS, rule-isolated configs.
- Root cause: `packages/*/src/core/ast-node/finder/public-property.ts` — `if (!property.typeAnnotation) return;` (both the `TSParameterProperty` and `PropertyDefinition` branches). The finder is shared with `no-construct-in-public-property-of-construct`; the annotation filter belongs in that rule, not in the shared finder.
- Note: `public bucket = new Bucket(this, "Bucket")` (initializer-inferred Construct type) is likewise invisible to `no-construct-in-public-property-of-construct` — worth deciding intentionally when touching the finder.
